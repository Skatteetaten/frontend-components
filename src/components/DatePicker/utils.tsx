export const DEFAULTFORMATDATE = (date: Date | null | undefined): string => {
  return date
    ? Intl.DateTimeFormat('no', {
        year: 'numeric',
        month: '2-digit',
        day: 'numeric',
      }).format(date)
    : '';
};

export const DEFAULTPARSEDATEFROMSTRING = (date: string): Date | null => {
  try {
    if (date && date.length === 10) {
      const day: number = parseInt(date.substring(0, 2));
      const month: number = parseInt(date.substring(3, 5));
      const year: number = parseInt(date.substring(6, 10));
      const formattedDate: Date = new Date(Date.UTC(year, month - 1, day));
      return formattedDate;
    }
  } catch (error) {
    console.log(error);
  }
  return null;
};

export const monthsForLocale = (
  localeName = 'no',
  monthFormat: 'long' | 'short' = 'long'
) => {
  const format = new Intl.DateTimeFormat(localeName, { month: monthFormat })
    .format;
  const months: string[] = [];
  for (let month = 0; month < 12; month++) {
    months.push(format(new Date(Date.UTC(2020, month))));
  }
  return months;
};

export const weekdaysForLocale = (
  localeName = 'no',
  dayFormat: 'long' | 'short' = 'long'
) => {
  const format = new Intl.DateTimeFormat(localeName, { weekday: dayFormat })
    .format;
  const weekdays: string[] = [];
  for (let day = 0; day < 7; day++) {
    weekdays.push(format(new Date(Date.UTC(2020, 7, day + 2)))); // add 2 days to let the week start on Sunday
  }
  return weekdays;
};
