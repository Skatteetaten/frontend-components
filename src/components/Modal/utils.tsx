import { useEffect } from 'react';

export const createModalDomPlacement = (
  setFlag: (a: boolean) => void,
  shadowRootNode: Document | ShadowRoot | undefined
) => {
  if (!shadowRootNode) {
    setFlag(true);
    return;
  }

  const styleInjectorWrapperId = 'style-injector';
  const modalWrapperId = 'modal-wrapper';

  if (shadowRootNode.getElementById(modalWrapperId)) {
    setFlag(true);
  } else {
    const modalWrapper = document.createElement('div');
    modalWrapper.id = modalWrapperId;
    const styleInjectorWrapper = shadowRootNode.getElementById(
      styleInjectorWrapperId
    );
    const parent = styleInjectorWrapper ? styleInjectorWrapper : shadowRootNode;
    parent.appendChild(modalWrapper);
    setFlag(true);
  }
};

export const useEscOnPress = (
  ref: HTMLDivElement | null,
  onClose?: (ref: HTMLDivElement) => void
): void => {
  useEffect(() => {
    const onKeyDown = (event: KeyboardEvent): void => {
      if (ref == null) {
        return;
      }
      if (event.code === 'Escape' && onClose) {
        onClose(ref);
      }
    };

    if (ref) {
      window.addEventListener('keydown', onKeyDown as EventListener, false);
    }
    return function cleanup(): void {
      if (window) {
        window.removeEventListener('keydown', onKeyDown as EventListener);
      }
    };
  }, [ref, onClose]);
};
