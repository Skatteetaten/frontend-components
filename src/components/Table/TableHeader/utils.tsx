import { TFunction } from 'react-i18next';

export const getIconNameForTheadCell = (
  isSorted: boolean,
  isSortedAscending: boolean
) => {
  if (isSorted) {
    return isSortedAscending ? 'ArrowDown' : 'ArrowUp';
  }

  return 'ArrowUpDown';
};

export const getAriaLabelForTheadCell = (
  name: string | JSX.Element,
  fieldName: string,
  isSorted: boolean,
  isSortedAscending: boolean,
  t: TFunction<Array<string>>
) => {
  let ariaLabel = fieldName;
  if (typeof name === 'string') {
    ariaLabel = isSorted
      ? isSortedAscending
        ? name.concat(' ', t('table.sorted_ascending'))
        : name.concat(' ', t('table.sorted_descending'))
      : name.concat(' ', t('table.sortable'));
  }

  return ariaLabel;
};
