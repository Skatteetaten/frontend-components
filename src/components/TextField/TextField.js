import React, { isValidElement } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import { Label } from 'office-ui-fabric-react/lib-commonjs/Label';
import { IconButton } from 'office-ui-fabric-react/lib-commonjs/Button';
//import { Callout, DirectionalHint } from 'office-ui-fabric-react/lib-commonjs/Callout';
import Callout from '../Callout/Callout';
import {
  TextField as FabricTextField,
  MaskedTextField as FabricMaskedTextField
} from 'office-ui-fabric-react/lib-commonjs/TextField';
import { getClassNames } from './TextField.classNames';
import { getClassNames as getLabelClassNames } from './TextFieldLabel.classNames';
import { isUndefined } from 'util';
import ErrorMessage from '../ErrorMessage/ErrorMessage';

class MaskedTextField extends FabricMaskedTextField {
  constructor(props) {
    super(props);
    this._skipComponentRefResolution = false;
  }
}
/**
 * @visibleName TextField (Tekstfelt)
 */
export default class TextField extends React.PureComponent {
  static propTypes = {
    /** Beskrivende informasjon til skjermlesere */
    ariaLabel: PropTypes.string,
    /** Automatisk utvide høyde (ved multiline) */
    autoAdjustHeight: PropTypes.bool,
    /** Benyttes når teksten for et readonly tekstfelt skal fremheves  */
    boldText: PropTypes.bool,
    /** Fjerne rammen rundt feltet */
    borderless: PropTypes.bool,
    /** Bestemmer om hjelptekst/varseltekst skal legge seg mellom label og tekstfelt eller flytende over innhold */
    calloutFloating: PropTypes.bool,
    /** For ytterligere stiling */
    className: PropTypes.string,
    /** Bestemmer om ett readonly felt skal være alltid redigerbart om det er tomt */
    editableWhenEmpty: PropTypes.bool,
    /** Benyttes når et readonly felt skal være redigertbart  */
    editable: PropTypes.bool,
    /** Tilhørende feilmelding */
    errorMessage: PropTypes.oneOfType([PropTypes.element, PropTypes.string]),
    /** Tilhørende hjelpetekst */
    help: PropTypes.oneOfType([PropTypes.element, PropTypes.string]),
    /** Størrelse på tekstfelt som skal benyttes */
    inputSize: PropTypes.oneOf(['normal', 'large']),
    /** Emnet over feltet */
    label: PropTypes.string,
    /** Størrelse på label */
    labelSize: PropTypes.oneOf(['small', 'large']),
    /** Felt med flere linjer */
    multiline: PropTypes.bool,
    /** Callback når input endres  */
    onChange: PropTypes.func,
    /** Tekst inni feltet som vises før man skriver */
    placeholder: PropTypes.string,
    /** Tekst inni feltet som alltid legges til foran, f.eks. www */
    prefix: PropTypes.string,
    /** Om feltet er inaktivt */
    readonly: PropTypes.bool,
    /** Tekst inni feltet som alltid legges til bak, f.eks. kr */
    suffix: PropTypes.string,
    /** Label på linje med verdi */
    underlined: PropTypes.bool,
    /** Tilhørende varseltekst */
    warning: PropTypes.oneOfType([PropTypes.element, PropTypes.string]),
    /** Formateringstreng som definerer oppførsel. Backslash vil escape et tegn. Spesialformat er:  '9': [0-9] 'a': [a-zA-Z] '*': [a-zA-Z0-9]. Eksempel tlf.nr 999 99 999 */
    mask: PropTypes.string,
    /** Antall rader som skal vises i feltet når multiline er satt */
    rows: PropTypes.number
  };

  static defaultProps = {
    inputSize: 'normal',
    editableWhenEmpty: false,
    labelSize: 'small',
    boldText: true,
    calloutFloating: false
  };

  constructor(props) {
    super(props);
    this.state = {
      isCalloutVisible: false,
      editMode: this.editMode(),
      value: '',
      activeCallout: '',
      lineBreak: false
    };
  }

  render() {
    const {
      children,
      onRenderLabel,
      readonly,
      disabled,
      inputSize,
      boldText,
      className,
      mask,
      ...props
    } = this.props;

    let TextFieldType = FabricTextField;
    let value = props.value;
    if (mask) {
      TextFieldType = MaskedTextField;
      value = value ? value : '';
    }

    const setValue = () => {
      if (this.props.suffix && this.props.readonly && !this.state.editMode) {
        return value + ' ' + this.props.suffix;
      }
      return value;
    };
    const { errorMessage, ...filteredProps } = this.props;

    return (
      <div className={classnames(getClassNames(this.props), className)}>
        <TextFieldType
          {...filteredProps}
          value={setValue()}
          ariaLabel={props.label}
          disabled={disabled}
          readOnly={this.state.editMode ? '' : readonly}
          className={classnames(
            getClassNames({ ...this.props, editMode: this.state.editMode }),
            className
          )}
          onRenderLabel={onRenderLabel || this._onRenderLabel}
          onBlur={this._onBlur}
          componentRef={inputField => {
            this.props.componentRef && this.props.componentRef(inputField);
            this._textFieldElement = inputField;
          }}
          inputSize={inputSize}
        >
          {children}
        </TextFieldType>
        {errorMessage && <ErrorMessage>{errorMessage}</ErrorMessage>}
      </div>
    );
  }

  _onRenderLabel = props => {
    const {
      label,
      help,
      warning,
      componentId,
      readOnly,
      editable,
      calloutFloating,
      inputSize
    } = props;
    const styles = getLabelClassNames(props);
    const inputSizeLarge = inputSize === 'large';

    let { isCalloutVisible, activeCallout } = this.state;
    const helpElement = isValidElement(help) ? help : <p>{help}</p>;

    let warningElement = isValidElement(warning) ? warning : <p>{warning}</p>;

    return (
      <div className={styles.labelArea}>
        <span className={styles.label}>
          {label ? <Label htmlFor={componentId}>{label}</Label> : null}
        </span>

        {readOnly && (
          <span className={styles.labelIconArea}>
            {editable && (
              <IconButton
                iconProps={{ iconName: 'Edit' }}
                title="Rediger"
                ariaLabel={'Rediger feltet ' + props.label}
                onClick={this._onEdit}
                className={styles.icon}
              />
            )}
          </span>
        )}

        {help && (
          <span
            className={styles.labelIconArea}
            ref={helpButton => (this._iconButtonElement = helpButton)}
          >
            <IconButton
              iconProps={{ iconName: 'HelpOutline' }}
              title="Hjelp"
              ariaLabel={'Hjelpetekst til feltet ' + props.label}
              onClick={() => this._onClick('helpCallout')}
              className={styles.icon}
            />
          </span>
        )}

        {warning && (
          <span
            className={styles.labelIconArea}
            ref={warningButton => (this._iconButtonElement = warningButton)}
          >
            <IconButton
              iconProps={{ iconName: 'WarningOutline' }}
              title="Varsel"
              ariaLabel={'Varseltekst til feltet ' + props.label}
              onClick={() => this._onClick('warningCallout')}
              className={styles.icon}
            />
          </span>
        )}

        {isCalloutVisible && (
          <Callout
            directionalHint={
              (inputSizeLarge && isUndefined(calloutFloating)) ||
              (!isUndefined(calloutFloating) && !calloutFloating)
                ? Callout.POS_BOTTOM_LEFT
                : Callout.POS_TOP_LEFT
            }
            color={
              activeCallout === 'helpCallout' ? Callout.HELP : Callout.WARNING
            }
            ariaLabel={
              activeCallout === 'helpCallout' ? 'Hjelpetekst' : 'Varseltekst'
            }
            target={this._iconButtonElement}
            onClose={this._onDismiss}
          >
            {activeCallout === 'helpCallout' ? helpElement : warningElement}
          </Callout>
        )}
      </div>
    );
  };

  _onClick = type => {
    let { isCalloutVisible, activeCallout } = this.state;

    if (activeCallout !== type && isCalloutVisible) {
      this.setState({
        activeCallout: type
      });
    } else {
      this.setState({
        activeCallout: type,
        isCalloutVisible: !isCalloutVisible
      });
    }
  };

  _onDismiss = () => {
    this.setState({
      isCalloutVisible: false
    });
  };

  _onEdit = () => {
    this._textFieldElement.focus();
    this.setState({ editMode: true });
  };

  _onBlur = e => {
    this.props.onBlur && this.props.onBlur(e);
    if (this.state.editMode === true) {
      this.setState({
        editMode: this.editMode()
      });
    }
  };

  editMode() {
    if (this.props.editableWhenEmpty && !this.props.value) {
      return this.props.value === '' || this.props.length === 0;
    }
    return false;
  }
}
