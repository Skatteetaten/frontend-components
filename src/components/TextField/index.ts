export { TextField as default } from './TextField';
export * from './TextField';
