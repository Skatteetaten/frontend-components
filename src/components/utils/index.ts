export { DefaultPalette as Palette, skeColor, skePalette } from './palette';
export type { PaletteProps, SkeColorProp } from './palette';
export {
  Fonts,
  FontSizes,
  LineHeightSizes,
  FontWeights,
  IconFontSizes,
  createFont,
} from './fonts';
export { omstrukturerFlatJSON, t } from './i18n/i18n';
export { getSrOnlyStyle } from './getSrOnlyStyle';
export { languages } from './i18n/languages';
export type { FontSizeType } from './fonts';
export { getFocusStyle } from './getFocusStyle';
export { Heading } from './Heading';
export { UseScreen } from './ScreenPlugin';
export { default as SkeIcons } from './icons/icons-ske';
export { Animation } from './getAnimationStyles';
export { takeIf } from './helpers';
export { generateId } from './generateId';
export { useHotkeys } from './useHotkeys';
