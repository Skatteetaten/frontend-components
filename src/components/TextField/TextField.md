** Tekstfelt benyttes når vi ønsker at brukeren skrive inn informasjon av en bestemt type - som oftest i et skjema **

Standard tekstfelt:

```js
import TextField from '@skatteetaten/frontend-components/TextField';

<div style={{ width: '300px' }}>
  <TextField
    id={'my-input-1'}
    label={'Navn'}
    placeholder={'For- og etternavn'}
    value={state.value1}
    onChange={(e, value) => setState({ value1: value })}
  />
</div>;
```

Stor variant av tekstfelt:

```js
import TextField from '@skatteetaten/frontend-components/TextField';

<div style={{ width: '300px' }}>
  <TextField
    id={'my-input-1'}
    label={'Navn'}
    inputSize={'large'}
    placeholder={'For- og etternavn'}
    value={state.value1}
    onChange={(e, value) => setState({ value1: value })}
  />
</div>;
```

Hjelpetekster og advarsel:

```js
import TextField from '@skatteetaten/frontend-components/TextField';

<>
  <div style={{ width: '300px' }}>
    <TextField
      id={'my-helpfield-1'}
      box
      withLeadingIcon="search"
      label="Fullt navn"
      calloutFloating={false}
      placeholder={''}
      value={state.value4}
      onChange={(e, value) => setState({ value4: value })}
      help="Vi trenger å vite navnet ditt dersom vi skal kontakte deg senere."
      calloutFloating={false}
    />
  </div>
  <br />
  <div style={{ width: '150px' }}>
    <TextField
      id={'my-helpfield-2'}
      box
      withLeadingIcon="search"
      label="Antall barn"
      placeholder={''}
      value={'23'}
      calloutFloating={false}
      warning="Er du sikker på at antall barn er riktig?"
      calloutFloating={false}
    />
  </div>
</>;
```

Feilmelding vises i umiddelbar nærhet til feltet:

```js
import TextField from '@skatteetaten/frontend-components/TextField';

<div style={{ width: '160px' }}>
  <TextField
    id={'my-errorfield-2'}
    box
    withLeadingIcon="search"
    label="Inntektsår"
    value={'0218'}
    errorMessage="Inntekståret må være etter 2008."
  />
</div>;
```

Av og til ønsker man å veksle mellom skrive og lesemodus. Da vises det som ren tekst med et blyant-ikon. Når man klikker, blir det skrivefelt:

```js
import TextField from '@skatteetaten/frontend-components/TextField';

const initialState = {
  value: 'Siri Saksbehandler',
  vekt: '4'
};

<div style={{ width: '300px' }}>
  <TextField
    id={'my-readonlyfield'}
    readOnly
    editable
    box
    label="Saksbehandler"
    value={state.value}
    onChange={(e, value) => setState({ value: value })}
    boldText={true}
  />
  <p>Med suffix:</p>
  <TextField
    id={'vektVare'}
    readOnly
    editable
    box
    label="Vekt på vare"
    value={state.vekt}
    onChange={(e, value) => setState({ vekt: value })}
    boldText={true}
    suffix={'kg'}
  />
</div>;
```

Bestemt formatering:

```js
import TextField from '@skatteetaten/frontend-components/TextField';

<div style={{ width: '300px' }}>
  <TextField
    id={'my-input-1'}
    onChange={(e, val) => console.log(val)}
    label={'Org.nummer'}
    placeholder={'999 999 999'}
    value={''}
    mask={'999 999 999'}
    maskChar={''}
  />
</div>;
```

Multiline tekstfelt:

```js
import TextField from '@skatteetaten/frontend-components/TextField';

<div style={{ width: '300px' }}>
  <TextField
    label="Endres til multiline dersom teksten er lengre enn 50 tegn"
    multiline={state.lineBreak}
    onChange={(e, value) => {
      const lineBreak = value.length > 50;
      if (lineBreak !== state.lineBreak) {
        setState({
          lineBreak: lineBreak
        });
      }
    }}
  />
  <br />
  <TextField
    label="Antall rader som vises i multiline modus kan settes"
    multiline
    rows={7}
  />
</div>;
```

```js noeditor
import Accordion from '@skatteetaten/frontend-components/Accordion';
import AccordionItem from '@skatteetaten/frontend-components/Accordion/AccordionItem';

<Accordion>
  <AccordionItem
    isOpen
    toggleContent
    toggleButtonText={'Bruk'}
    stepId={'step-1-1'}
  >
    <p>
      I Designsystemet finnes det mange ulike varisjoner av tekstfelt. Vi har
      med og uten hjelpetekst, feilmeldinger eller lese- og skrivemodus. Vi har
      også mulighet til å bestemme formatering, for eksempel ved
      organiasasjonsnummer.
    </p>
  </AccordionItem>
  <AccordionItem
    toggleContent
    toggleButtonText={'Universell utforming'}
    stepId={'step-1-2'}
  >
    <p>
      Det skal alltid være en programatisk kobling mellom tekstfelt og label,
      slik at skjermleser kan lese opp beskrivelsen av feltet når det er i
      fokus.
    </p>
    <p>
      I publikumsløsninger viser i advarsel- og hjelpetekst mellom label og
      verdi (sett calloutFloating til false). Hjelpetekster i Callout fanges
      ikke like enkelt opp av skjermlesere.
    </p>
  </AccordionItem>
  <AccordionItem
    toggleContent
    toggleButtonText={'Tekniske detaljer'}
    stepId={'step-1-3'}
  >
    <p>
      <a
        href="https://developer.microsoft.com/en-us/fabric#/components/textfield#Implementation"
        target="_blank"
      >
        Se flere tilgjengelige props i Fabric dokumentasjonen
      </a>
    </p>
  </AccordionItem>
</Accordion>;
```
