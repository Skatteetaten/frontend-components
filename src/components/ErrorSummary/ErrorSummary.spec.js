import React from 'react';
import { mount } from 'enzyme';
import { ErrorSummary } from '.';
import { TextField } from '../TextField';
import { matches, DummyWebComp } from '../utils/test-utils';

function oppsettMount(props, children) {
  window.HTMLElement.prototype.scrollIntoView = jest.fn();
  const holder = document.createElement('div');
  document.body.appendChild(holder);
  return mount(
    <div>
      <TextField
        id={'input_adresse'}
        label={'Adresse'}
        errorMessage={'Du må legge til en adresse'}
      />
      <TextField
        id={'input_postkode'}
        label={'Postnummer'}
        errorMessage={'Du må legge til en postkode'}
      />
      <TextField
        id={'input_poststed'}
        label={'Poststed'}
        errorMessage={'Du må legge til et poststed'}
      />
      <ErrorSummary {...props}>{children}</ErrorSummary>
    </div>,
    { attachTo: holder }
  );
}

describe('ErrorSummary komponent', () => {
  it('matcher snapshot', () => {
    matches(
      <ErrorSummary
        title={'For å gå videre må du rette opp i følgende:'}
        errors={[
          { id: 'input_adresse-input', error: 'Du må legge til en adresse' },
          {
            id: 'input_postkode-input',
            error: 'Du må legge til en postkode',
          },
          {
            id: 'input_poststed-input',
            error: 'Du må legge til et poststed',
          },
        ]}
      />
    );
  });

  it('Skal kalle onClick dersom den er angitt', async () => {
    const onClickMock = jest.fn();
    const wrapper = oppsettMount({
      onClick: onClickMock,
      title: 'For å gå videre må du rette opp i følgende:',
      errors: [
        { id: 'input_adresse-input', error: 'Du må legge til en adresse' },
        {
          id: 'input_postkode-input',
          error: 'Du må legge til en postkode',
        },
        {
          id: 'input_poststed-input',
          error: 'Du må legge til et poststed',
        },
      ],
    });

    wrapper.find('li').at(1).find('a').simulate('click');
    expect(onClickMock).toHaveBeenCalledTimes(1);
  });

  it('Setter fokus på riktig inputfelt når link blir trykket på og at onClick ikke er angitt', () => {
    const wrapper = oppsettMount({
      title: 'For å gå videre må du rette opp i følgende:',
      errors: [
        { id: 'input_adresse-input', error: 'Du må legge til en adresse' },
        {
          id: 'input_postkode-input',
          error: 'Du må legge til en postkode',
        },
        {
          id: 'input_poststed-input',
          error: 'Du må legge til et poststed',
        },
      ],
    });

    expect(wrapper.find('h3').text()).toEqual(
      'For å gå videre må du rette opp i følgende:'
    );
    wrapper.find('li').at(1).find('a').simulate('click');
    expect(document.activeElement.id).toBe(
      wrapper.find('input').at(1).getDOMNode().id
    );
    wrapper.find('li').at(2).find('a').simulate('click');
    expect(document.activeElement.id).toBe(
      wrapper.find('input').at(2).getDOMNode().id
    );
  });

  it('Setter fokus på inputfelt i egen shadowRoot når link blir trykket på og at shadowRootNode er angitt', async () => {
    const original = global.document['body'];
    window.customElements.define('webcomp-test', DummyWebComp);
    document.body.innerHTML = '<webcomp-test></webcomp-test>';

    const holder = document.createElement('div');

    const element = document.querySelector('webcomp-test');
    element.shadowRoot.appendChild(holder);

    const wrapper = mount(
      <div>
        <TextField
          id={'input_postkode'}
          label={'Postnummer'}
          errorMessage={'Du må legge inn postnummer.'}
        />
        <ErrorSummary
          shadowRootNode={element.shadowRoot}
          title={'For å gå videre må du rette opp i følgende'}
          errors={[
            { id: 'input_adresse-input', error: 'Du må legge til en adresse' },
            {
              id: 'input_postkode-input',
              error: 'Du må legge til en postkode',
            },
            {
              id: 'input_poststed-input',
              error: 'Du må legge til et poststed',
            },
          ]}
        />
      </div>,
      { attachTo: holder }
    );
    wrapper.find('li').at(1).find('a').simulate('click');
    expect(element.shadowRoot.activeElement.id).toBe(
      wrapper.find('input').getDOMNode().id
    );
    global.document['body'] = original;
  });

  it('Skal ikke vise ErrorSummary hvis ingen feil eller children finnes', () => {
    const wrapper = oppsettMount({
      title: 'For å gå videre må du rette opp i følgende:',
      errors: [],
    });
    expect(wrapper.find('h3').exists()).toBeFalsy();
    expect(wrapper.find('li').exists()).toBeFalsy();
  });

  it('Skal ikke vise ErrorSummary hvis errors er undefined', () => {
    const wrapper = oppsettMount({
      title: 'For å gå videre må du rette opp i følgende:',
      errors: undefined,
    });
    expect(wrapper.find('h3').exists()).toBeFalsy();
    expect(wrapper.find('li').exists()).toBeFalsy();
  });

  it('Skal vise children hvis de finnes', () => {
    const wrapper = oppsettMount(
      {
        title: 'For å gå videre må du rette opp i følgende:',
        errors: undefined,
      },
      <p>test child</p>
    );

    expect(wrapper.find('p').text()).toBe('test child');
  });
});
