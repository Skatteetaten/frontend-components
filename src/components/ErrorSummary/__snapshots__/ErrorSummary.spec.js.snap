// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorSummary komponent matcher snapshot 1`] = `
<div
  aria-live="assertive"
  className=

      {
        background-color: #ffffff;
        border: 2px solid #a82e39;
        display: flex;
        justify-content: flex-start;
        width: fit-content;
      }
      & h1 {
        font-size: 1.125rem;
        margin-bottom: 0;
        margin-left: 0;
        margin-right: 0;
        margin-top: 0;
      }
      & h2 {
        font-size: 1.125rem;
        margin-bottom: 0;
        margin-left: 0;
        margin-right: 0;
        margin-top: 0;
      }
      & h3 {
        font-size: 1.125rem;
        margin-bottom: 0;
        margin-left: 0;
        margin-right: 0;
        margin-top: 0;
      }
      & h4 {
        font-size: 1.125rem;
        margin-bottom: 0;
        margin-left: 0;
        margin-right: 0;
        margin-top: 0;
      }
      & h5 {
        font-size: 1.125rem;
        margin-bottom: 0;
        margin-left: 0;
        margin-right: 0;
        margin-top: 0;
      }
      & h6 {
        font-size: 1.125rem;
        margin-bottom: 0;
        margin-left: 0;
        margin-right: 0;
        margin-top: 0;
      }
      & ul {
        padding-bottom: 0px;
        padding-left: 0px;
        padding-right: 0px;
        padding-top: 0px;
      }
      & li {
        cursor: pointer;
        display: block;
        line-height: 1.636;
      }
  id="errorsummary"
  role="status"
>
  <div
    className=

        {
          background-color: #a82e39;
          text-align: center;
          width: 4rem;
        }
        @media (max-width: 639px){& {
          width: 2rem;
        }
  >
    <div
      className=

          {
            color: #ffffff;
            font-size: 2.4rem;
            padding-bottom: 8x;
            padding-left: 4px;
            padding-right: 8px;
            padding-top: 6px;
          }
          @media (max-width: 639px){& {
            font-size: 1.4rem;
          }
    >
      <i
        aria-hidden={true}
        className=

            {
              -moz-osx-font-smoothing: grayscale;
              -webkit-font-smoothing: antialiased;
              display: inline-block;
              font-family: "Material Icons";
              font-style: normal;
              font-weight: normal;
              speak: none;
            }
        data-icon-name="Error"
      >
        
      </i>
    </div>
  </div>
  <div
    className=

        {
          padding-bottom: 16px;
          padding-left: 24px;
          padding-right: 24px;
          padding-top: 16px;
        }
        @media (max-width: 639px){& {
          padding-bottom: 8px;
          padding-left: 12px;
          padding-right: 12px;
          padding-top: 8px;
        }
  >
    <h3>
      For å gå videre må du rette opp i følgende:
    </h3>
    <ul>
      <li>
        <span
          className=

              {
                margin-bottom: 0px;
                margin-left: 0px;
                margin-right: 0px;
                margin-top: 0px;
              }
              &:hover>i {
                color: #093e61;
                padding-left: 4px;
                padding-right: 6px;
                transition: 0.1s;
              }
              &:hover>a {
                background-color: #cde1f9;
              }
        >
          <i
            aria-hidden={true}
            className=

                {
                  -moz-osx-font-smoothing: grayscale;
                  -webkit-font-smoothing: antialiased;
                  border-bottom: none;
                  color: #1362ae;
                  display: inline-block;
                  font-family: "SkeIconfont";
                  font-size: 1rem;
                  font-style: normal;
                  font-weight: 700;
                  padding-left: 0px;
                  padding-right: 10px;
                  speak: none;
                  text-rendering: auto;
                  transform: translate(0, 0);
                  vertical-align: bottom;
                }
            data-icon-name="ArrowForward"
            role="presentation"
          >
            
          </i>
          <a
            className=

                {
                  border-bottom: 1px solid #1362ae;
                  color: #1362ae;
                  font-weight: 700;
                  text-decoration: none;
                }
                &:hover {
                  background-color: #cde1f9;
                  border-bottom: 2px solid #093e61;
                  color: #093e61;
                }
                &:focus {
                  background-color: #cde1f9;
                  border-bottom: 2px solid #093e61;
                  color: #093e61;
                  outline: none;
                }
            onClick={[Function]}
            tabIndex={0}
          >
            Du må legge til en adresse
          </a>
        </span>
      </li>
      <li>
        <span
          className=

              {
                margin-bottom: 0px;
                margin-left: 0px;
                margin-right: 0px;
                margin-top: 0px;
              }
              &:hover>i {
                color: #093e61;
                padding-left: 4px;
                padding-right: 6px;
                transition: 0.1s;
              }
              &:hover>a {
                background-color: #cde1f9;
              }
        >
          <i
            aria-hidden={true}
            className=

                {
                  -moz-osx-font-smoothing: grayscale;
                  -webkit-font-smoothing: antialiased;
                  border-bottom: none;
                  color: #1362ae;
                  display: inline-block;
                  font-family: "SkeIconfont";
                  font-size: 1rem;
                  font-style: normal;
                  font-weight: 700;
                  padding-left: 0px;
                  padding-right: 10px;
                  speak: none;
                  text-rendering: auto;
                  transform: translate(0, 0);
                  vertical-align: bottom;
                }
            data-icon-name="ArrowForward"
            role="presentation"
          >
            
          </i>
          <a
            className=

                {
                  border-bottom: 1px solid #1362ae;
                  color: #1362ae;
                  font-weight: 700;
                  text-decoration: none;
                }
                &:hover {
                  background-color: #cde1f9;
                  border-bottom: 2px solid #093e61;
                  color: #093e61;
                }
                &:focus {
                  background-color: #cde1f9;
                  border-bottom: 2px solid #093e61;
                  color: #093e61;
                  outline: none;
                }
            onClick={[Function]}
            tabIndex={0}
          >
            Du må legge til en postkode
          </a>
        </span>
      </li>
      <li>
        <span
          className=

              {
                margin-bottom: 0px;
                margin-left: 0px;
                margin-right: 0px;
                margin-top: 0px;
              }
              &:hover>i {
                color: #093e61;
                padding-left: 4px;
                padding-right: 6px;
                transition: 0.1s;
              }
              &:hover>a {
                background-color: #cde1f9;
              }
        >
          <i
            aria-hidden={true}
            className=

                {
                  -moz-osx-font-smoothing: grayscale;
                  -webkit-font-smoothing: antialiased;
                  border-bottom: none;
                  color: #1362ae;
                  display: inline-block;
                  font-family: "SkeIconfont";
                  font-size: 1rem;
                  font-style: normal;
                  font-weight: 700;
                  padding-left: 0px;
                  padding-right: 10px;
                  speak: none;
                  text-rendering: auto;
                  transform: translate(0, 0);
                  vertical-align: bottom;
                }
            data-icon-name="ArrowForward"
            role="presentation"
          >
            
          </i>
          <a
            className=

                {
                  border-bottom: 1px solid #1362ae;
                  color: #1362ae;
                  font-weight: 700;
                  text-decoration: none;
                }
                &:hover {
                  background-color: #cde1f9;
                  border-bottom: 2px solid #093e61;
                  color: #093e61;
                }
                &:focus {
                  background-color: #cde1f9;
                  border-bottom: 2px solid #093e61;
                  color: #093e61;
                  outline: none;
                }
            onClick={[Function]}
            tabIndex={0}
          >
            Du må legge til et poststed
          </a>
        </span>
      </li>
    </ul>
  </div>
</div>
`;
